src/serac/physics/tests/serac_thermal_functional.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "serac/physics/thermal_conduction_functional.hpp"
+
+#include <fstream>
+
+#include <gtest/gtest.h>
+#include "mfem.hpp"
+
+#include "serac/mesh/mesh_utils.hpp"
+#include "serac/physics/state/state_manager.hpp"
+#include "serac/serac_config.hpp"
+
+namespace serac {
+
+template <int p, int dim>
+void functional_test()
+{
+  MPI_Barrier(MPI_COMM_WORLD);
+
+  int serial_refinement   = 1;
+  int parallel_refinement = 0;
+
+  std::string meshfile2D = SERAC_REPO_DIR "/data/meshes/star.mesh";
+  auto        mesh2D = mesh::refineAndDistribute(buildMeshFromFile(meshfile2D), serial_refinement, parallel_refinement);
+  mesh2D->ExchangeFaceNbrData();
```

Comment:
I think `mfem::ParMesh::ExchangeFaceNbdrData()` is already being called here:
https://github.com/LLNL/serac/blob/d7cce1d1b788e7e859d21ad0559ccb741c501c8e/src/serac/mesh/mesh_utils.cpp#L433